MapboxDirectionsTests/IntructionsTests.swift

```diff
@@ -84,21 +84,90 @@ class SpokenInstructionsTests: XCTestCase {
-        XCTAssertEqual(visualInstructions?.first?.primaryInstruction.textComponents.first?.abbreviationPriority, 0)
+        XCTAssertEqual(visualInstructionComponent.type, .text)
+        XCTAssertEqual(visualInstructionComponent.abbreviation, "Page St")
+        XCTAssertEqual(visualInstructionComponent.abbreviationPriority, 0)
         XCTAssertEqual(visualInstructions?.first?.drivingSide, .right)
         XCTAssertNil(visualInstructions?.first?.secondaryInstruction)
         
         let arrivalVisualInstructions = arrivalStep.instructionsDisplayedAlongStep!
         XCTAssertEqual(arrivalVisualInstructions.first?.secondaryInstruction?.text, "the gym")
     }
+    
+    func testTertiaryInstructions() {
+        let expectation = self.expectation(description: "calculating directions with tertiary instructions should return results")
+        let queryParams: [String: String?] = [
+            "geometries": "polyline",
+            "steps": "true",
+            "access_token": BogusToken,
+            "banner_instructions": "true"
+        ]
+        
+        stub(condition: isHost("api.mapbox.com") && containsQueryParams(queryParams)) { _ in
+            let path = Bundle(for: type(of: self)).path(forResource: "subInstructionsWithLanes", ofType: "json")
+            return OHHTTPStubsResponse(fileAtPath: path!, statusCode: 200, headers: ["Content-Type": "application/json"])
+        }
+        
+        let startWaypoint =  Waypoint(coordinate: CLLocationCoordinate2D(latitude: 39.132063, longitude: -84.531074))
+        let endWaypoint = Waypoint(coordinate: CLLocationCoordinate2D(latitude: 39.138953, longitude: -84.532934))
+        
+        let options = RouteOptions(waypoints: [startWaypoint, endWaypoint], profileIdentifier: .automobileAvoidingTraffic)
+        options.shapeFormat = .polyline
+        options.includesSteps = true
+        options.includesAlternativeRoutes = false
+        options.includesVisualInstructions = true
+        
+        var route: Route?
+        let task = Directions(accessToken: BogusToken).calculate(options) { (waypoints, routes, error) in
+            XCTAssertNil(error, "Error: \(error!.localizedDescription)")
+            
+            XCTAssertNotNil(routes)
+            XCTAssertEqual(routes!.count, 1)
+            route = routes!.first!
+            
+            expectation.fulfill()
+        }
+        XCTAssertNotNil(task)
+        
+        waitForExpectations(timeout: 2) { (error) in
+            XCTAssertNil(error, "Error: \(error!.localizedDescription)")
+            XCTAssertEqual(task.state, .completed)
+        }
+        
+        XCTAssertNotNil(route)
+        XCTAssertEqual(route!.routeIdentifier, "cjikck25m00v279ms1knttdgc")
+        
+        let step = route!.legs.first!.steps.first!
+        let visualInstructions = step.instructionsDisplayedAlongStep
+        
+        let tertiaryInstruction = visualInstructions?.first?.tertiaryInstruction
```

Comment:
@vincethecoder getting close! I think the final piece is to now add a test for a tertiaryInstruction where it's properties are non-nil. IE test for a then instruction instead of a lane instruction.

Reply:
@bsudekum accounted for here: https://github.com/mapbox/MapboxDirections.swift/pull/258/files#diff-c645476c0ab60998ed8e475352ee8bdcR174